### ISSUE TITLE
gogrepoc failing to pull .exe files

### ISSUE THREAD

**Issue by drijen** [Kalanyr/gogrepoc#114]
OS: Ubuntu 24.04.1 LTS
Python 3.12.3-0ubuntu2 

Running gogrepoc will result in a manifest that includes all files *except* the actual game
 .exe or any other non linux executable. Shell scripts for linux compatible games will be downloaded, but not their windows counterparts.

Example game: blake_stone_aliens_of_gold, linux and windows compatible; gogrepoc pulls *.sh, but not *.exe.
Example game 2: baldurs_gate_iii, windows only; no files downloaded aside from "extras". 

Have attached manifest and log, please let me know what else I can do to assist.

* Edited by Kalanyr - To Remove Logs for Privacy since they are no longer needed *

**Comment by Kalanyr** [comment]
What commands are you using ?
The default behaviour on Linux is to update/download only Linux files +
Extras unless something else is specificed with -os.

**Comment by drijen** [comment]
Ah, I see my error now.

Before, I took the default os selection behavior to be "all". 
Executing 'python3 gogrepoc.py update -os windows -os linux' resolved my issue.

I apologize, and also - thanks for maintaining this tool; really appreciate it.

**Comment by danielaixer** [comment]
@drijen I think you may want to remove your email from the log file, for privacy reasons...

**Comment by danielaixer** [comment]
Also, related to the issue's theme:

DO NOT forget to also specify -lang as well as -os. I've been debugging the script for over an hour because I wasn't able to pull the installers, even after specifying the OSes:
`python gogrepoc.py update -os windows linux -full`
This was giving me only the extras...

After investigating how the `filter_downloads` function works and checking which `lang_list` it was working with, if found out that this was the required command:
`python gogrepoc.py update -os windows linux -lang en es -full`

I'm using the -full parameter just in case.

@Kalanyr I would strongly recommend updating the "Quick Start -- Typical Use Case" readme section, by including there the -os or -lang parameters (not only under "Advanced Usage -- Common Tasks") or by adding a clear disclaimer indicating that `gogrepoc.py update`,etc. filter by the current OS **and** OS' language.

**Comment by Kalanyr** [comment]
Yes, I'll make a note to add that the default behaviour is to download only the current OS and current system language ( this seems to be the behaviour that's generally expected but it does get weird if you're running the script on a *nix NAS and are generally a Windows users or if you own games not in your System language ( which is probably way more common for languages other than English ) .

**Comment by just-Bri** [comment]
> this seems to be the behaviour that's generally expected but it does get weird if you're running the script on a *nix NAS and are generally a Windows users

This is me haha   
Was wondering why I wasn't seeing exe's in the `-dryrun`, ended up here.  
Thanks @danielaixer for posting the commands you used!

**Comment by fenuks** [comment]
Still doesn't work for me. Running `python gogrepoc.py download backup -lang en -os windows linux` Windows installers are not downloaded. Linux installers are fetched, but e.g. for game `XIII` that has no Linux version, nothing is downloaded. I also tried `-skipos mac`, but result is the same.

**Comment by Kalanyr** [comment]
Have you done an update for the game using those parameters before
attempting to download ? If your manifest was built with the default
settings it won't have any windows binaries in there to download.

**Comment by fenuks** [comment]
Thank you! I wasn't aware that manifests are affected by these flags. I thought
that lang/os filtering is done at the download phase. After refreshing
manifests, exe files are being downloaded.
